www-servers/lighttpd: add 1.4.79

gstrauss opened the pull request:
www-servers/lighttpd: add 1.4.79

Note: this release of lighttpd includes a hardened `doc/systemd/lighttpd.service`, but gentoo has its own independent `lighttpd.service` file.  I have not modified the gentoo lighttpd.service in this PR.  Please review the upstream `doc/systemd/lighttpd.service` and advise if you would like me to make those changes here or not.  Thanks!

reference: https://git.lighttpd.net/lighttpd/lighttpd1.4/src/branch/master/doc/systemd/lighttpd.service

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [ ] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

thesamesam commented:
Hi! Thank you as ever.

I think the changes upstream look good -- would you mind importing them (in a new revision of the service file, so as not to affect old versions of lighttpd in-tree)?

(That said, I am not actually sure _why_ we have our own .service file, or if that's some obsolete leftover...)

gstrauss commented:
> I think the changes upstream look good -- would you mind importing them (in a new revision of the service file, so as not to affect old versions of lighttpd in-tree)?

I updated the PR with new file `lighttpd.service-r3`, and updated `lighttpd-1.4.79.ebuild` and `lighttpd-9999.ebuild`
The only substantive difference between `lighttpd.service-r3` and upstream `lighttpd.service` is that the Gentoo lighttpd.service* has
`After=syslog.target network-online.target`
while upstream has
`After=network-online.target`

I preserved the Gentoo modification in `lighttpd.service-r3`.  While the default lighttpd.conf does not use syslog, syslog is an option and `syslog.target` seems like a harmless additional startup dependency.

gstrauss commented:
FYI: lighttpd.service-r1 was added 16 May 2024 in c31dd78225219d808259cd4ee04d25ba8a1142de so the history appears to be that you and/or I added `www-servers/lighttpd/files/lighttpd.service-r1` not so long ago.

thesamesam commented:
Seems the unit was added some time ago in https://bugs.gentoo.org/466254. It's a bit strange as it looks like lighttpd already had a unit upstream at that time.

gstrauss commented:
Thanks @thesamesam!

According to that link, the lighttpd.service was taken from Fedora at the time.  Makes sense.  Fedora just started using the lighttpd.service from upstream with this release (lighttpd 1.4.79), too.
